app/src/main/java/org/p2p/wallet/newsend/NewSendFragment.kt

```diff
@@ -46,18 +51,45 @@ class NewSendFragment :
             setSwitchLabel(
                 getString(R.string.send_switch_to_token, Constants.SOL_SYMBOL)
             )
+            tokenClickListener = {
+                presenter.onTokenClicked()
+            }
         }
         // TODO PWN-6090 make button
         binding.sliderSend.setActionText(R.string.send_enter_amount)
+
+        requireActivity().supportFragmentManager.setFragmentResultListener(
+            KEY_REQUEST_SEND,
+            viewLifecycleOwner
+        ) { _, result -> handleSupportFragmentResult(result) }
     }
 
-    // TODO PWN-6090 connect this method to token selection
-    fun showSourceToken(token: Token.Active) {
+    private fun handleSupportFragmentResult(result: Bundle) {
+        when {
+            // will be more!
+            result.containsKey(EXTRA_TOKEN) -> {
+                val token = result.getParcelable<Token.Active>(EXTRA_TOKEN)
+                if (token != null) presenter.setSourceToken(token)
+            }
+        }
+    }
+
+    override fun showSourceToken(token: Token.Active) {
```

Comment:
i don't think Source is a good naming in context of Send
in Swap, yes, we have source token and destination token which we want to swap, but in Send we only have token to send.
what do you think about showTokenToSend?

Reply:
ok maybe